docs-src/design-decisions.md

```diff
@@ -0,0 +1,42 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Better modeling for implementations with multiple GatewayClass resources
+  (including some potential multi-tenant scenarios using Cluster API).
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
+  in the same namespace by default).
+* GatewayClass parameters can live in the same namespace.
+* More RBAC flexibility (can allow users to create GatewayClasses in select
+  namespaces only or restrict visibility of GatewayClasses in some namespaces).
+
+### Advantages of a Cluster-Scoped GatewayClass
+* Consistent with StorageClass and IngressClass resources.
+* Discussions with sig-storage showed that StorageClass had worked well as a
```

Comment:
Did that discussion bring up the topic of managing other clusters with the operator pattern enabled by Cluster API?

Reply:
It did not, maybe @saad-ali has some insight here on if that would change anything from the perspective of sig-storage?

Reply:
You'll have to expound on the specifics of what "managing" means to you.